src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/ContractABI/index.tsx

```diff
@@ -19,12 +23,40 @@ const mustBeValidABI = (abi: string): undefined | string => {
   }
 }
 
-const ContractABI = () => (
-  <Row margin="sm">
-    <Col>
-      <TextareaField name="abi" placeholder="ABI*" text="ABI*" type="text" validate={mustBeValidABI} />
-    </Col>
-  </Row>
-)
+const ContractABI = (): React.ReactElement => {
+  const {
+    input: { value: contractAddress },
+  } = useField('contractAddress', { subscription: { value: true } })
+  const { mutators } = useForm()
+  const setAbiValue = React.useRef(mutators.setAbiValue)
+
+  React.useEffect(() => {
+    const validateAndSetAbi = async () => {
+      if (!mustBeEthereumAddress(contractAddress) && !(await mustBeEthereumContractAddress(contractAddress))) {
+        const network = getNetwork()
+        const source = getConfiguredSource()
+        const abi = await source.getContractABI(contractAddress, network)
+
+        // this check may help in scenarios where the user first pastes the ABI,
+        // and then sets a Proxy contract that has no useful methods
+        if (hasUsefulMethods(abi) === undefined) {
```

Comment:
`hasUsefulMethods` should return a boolean no be consistent, otherwise, it should be `getUsefulMethods`.

Reply:
yes, totally